cloudmarker/util.py

```diff
@@ -32,6 +33,51 @@ def load_config(config_paths):
     return config
 
 
+def load_plugin(plugin_config):
+    """Construct an object with specified plugin class and parameters.
+
+    The plugin_config parameter must be a dictionary with the following
+    keys:
+
+      - 'plugin': The value for this key must be a string that
+        represents the fully qualified class name of the plugin. The
+        fully qualified class name is in the dotted notation, e.g.,
+        ``pkg.module.ClassName``.
+      - 'params': The value for this key must be a dict that represents
+        the parameters to be passed to the ``__init__`` functio of the
+        plugin class. Each key in the dictionary represents the
+        parameter name and each value represents the value of the
+        parameter.
+
+    Arguments:
+        plugin_config (dict): Plugin configuration dictionary.
+
+    Returns:
+        object: An object of type mentioned in the ``plugin`` parameter.
+
+    """
+    # Split the fully qualified class name into module and class names.
+    parts = plugin_config['plugin'].rsplit('.', 1)
+
+    # Validate that the fully qualified class name had at least two
+    # parts: module name and class name.
+    if len(parts) < 2:
+        msg = ('Invalid plugin class name: {}; expected format: '
+               '[<pkg>.]<module>.<class>'.format(plugin_config['plugin']))
+        raise PluginError(msg)
+
+    # Load the specified adapter class from the specified module.
+    plugin_module = importlib.import_module(parts[0])
+    plugin_class = getattr(plugin_module, parts[1])
+
```

Comment:
I think this part of code should raise an error if the class name looks valid but does not exist. There could be a corresponding test case for it.

Reply:
Yes, it would lead to `AttributeError` because the attribute (the class we are looking for) would be missing from the module. I decided not to test this earlier because it is akin to testing whether the Python language is working properly or not.

This is one of those gray areas where I find it difficult to decide whether *to test or not to test*.

Now that you raise this point, I have now decided to add a test for it with the intention of *documenting the expected behavior as a test* in case of a missing plugin.